src/analyses/base.ml

```diff
@@ -1324,7 +1324,7 @@ struct
     let inv_bin_int (a, b) ikind c op =
       let warn_and_top_on_zero x =
         if GU.opt_predicate (BI.equal BI.zero) (ID.to_int x) then
-          (M.warn "Must Undefined Behavior: Second argument of div or mod is 0, continuing with top";
+          (M.warn (M.LogEvent.may (M.EventType.Unknown ("Must Undefined Behavior: Second argument of div or mod is 0, continuing with top")));
```

Comment:
What's the point of defining `Certainty.Must` and `BehaviorEvent` if they're not properly used? Same goes for many other `EventType`s.

Reply:
We have currently only changed our newly added warnings to their proper types. These changes with the unknowns are for backwards compatibility to make it compile as we were not sure what exact types should be which of these lines. This line you commented on is especially tricky since the text says _Undefined behavior_ but I think in the categories it should be _Integer.DivByZero_ ?

Reply:
> it should be `Integer.DivByZero`

Which is undefined behavior (same goes for `Integer.Overflow`).
Those `Event` modules need some cleanup/reordering.
If each of those modules just consists of a `type t` and `show` for it, you could also just put all types in one module and name the types accordingly.